backend/FwLite/FwDataMiniLcmBridge/Api/FwDataMiniLcmApi.cs

```diff
@@ -796,12 +821,58 @@
     /// <summary>
     /// must be called as part of an lcm action
     /// </summary>
-    internal void AddComplexFormComponent(ILexEntry lexEntry, ComplexFormComponent component)
+    internal void AddComplexFormComponent(ILexEntry lexComplexForm, ComplexFormComponent component, BetweenPosition<ComplexFormComponent>? between = null)
     {
         ICmObject lexComponent = component.ComponentSenseId is not null
             ? SenseRepository.GetObject(component.ComponentSenseId.Value)
             : EntriesRepository.GetObject(component.ComponentEntryId);
-        lexEntry.AddComponent(lexComponent);
+        InsertComplexFormComponent(lexComplexForm, lexComponent, between);
+    }
+
+    internal void InsertComplexFormComponent(ILexEntry lexComplexForm, ICmObject lexComponent, BetweenPosition<ComplexFormComponent>? between = null)
+    {
+        var previousComponentId = between?.Previous?.ComponentSenseId ?? between?.Previous?.ComponentEntryId;
+        var nextComponentId = between?.Next?.ComponentSenseId ?? between?.Next?.ComponentEntryId;
+
+        var entryRef = lexComplexForm.ComplexFormEntryRefs.SingleOrDefault();
+        if (entryRef is null || entryRef.ComponentLexemesRS.Count == 0)
+        {
+            lexComplexForm.AddComponent(lexComponent);
+            return;
+        }
+
+        // Prevents adding duplicates (which ComponentLexemesRS.Insert is susceptible to)
+        if (entryRef.ComponentLexemesRS.Contains(lexComponent))
+        {
+            if (between is null) return;
+            entryRef.ComponentLexemesRS.Remove(lexComponent);
+        }
+
+        var previousComponent = previousComponentId.HasValue ? entryRef.ComponentLexemesRS.FirstOrDefault(s => s.Guid == previousComponentId) : null;
```

Comment:
I think this line and another similar one below can be simplified to this
```suggestion
        var previousComponent = entryRef.ComponentLexemesRS.FirstOrDefault(s => s.Guid == previousComponentId);
```
no need to check if it has a value first as null will just fail the comparison always